src/crossover.rs

```diff
@@ -55,4 +62,60 @@ impl Crossover {
     pub fn encrypted_data(&self) -> &PoseidonCipher {
         &self.encrypted_data
     }
+
+    /// Converts a Crossover into it's byte representation
+    pub fn to_bytes(&self) -> [u8; Crossover::serialized_size()] {
+        let mut buf = [0u8; Crossover::serialized_size()];
+        let mut n = 0;
+
+        buf[n..n + 32].copy_from_slice(
+            &JubJubAffine::from(&self.value_commitment).to_bytes()[..],
+        );
+        n += 32;
+
+        buf[n..n + 32].copy_from_slice(&self.nonce.to_bytes()[..]);
+        n += 32;
+
+        buf[n..n + PoseidonCipher::cipher_size_bytes()]
+            .copy_from_slice(&self.encrypted_data.to_bytes()[..]);
+        n += PoseidonCipher::cipher_size_bytes();
+
+        assert_eq!(n, Crossover::serialized_size());
+
+        buf
+    }
+
+    /// Attempts to convert a byte representation of a note into a `Note`,
+    /// failing if the input is invalid
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
+        if bytes.len() < Crossover::serialized_size() {
+            return Err(Error::InvalidCrossoverConversion);
```

Comment:
It's okay because you've been consistent within the repos, but I think we start to have more clear errors for such scenarios: "InvalidXXXConversion" doesn't tell why. Here the error is that we don't have enough bytes into the buffer for actually even trying a conversion.

You can search for similar code in our repos, and get an inspiration for what I mean, we might want to refactor this also in the future.

Reply:
We can file an issue for it. It would be nice to be consistent although I don't think we should give prio to this atm.